### PR TITLE
port some more functions

### PR DESCRIPTION
**Pull request by brotzeit**

### CODE REVIEW

File: rust_src/src/editfns.rs [hunk 1]
```diff
@@ -132,3 +133,22 @@ pub fn point_min() -> LispObject {
...
+
+/// Set point to POSITION, a number or marker.
+/// Beginning of buffer is position (point-min), end is (point-max).
+///
+/// The return value is POSITION.
+#[lisp_fn]
+pub fn goto_char(position: LispObject) -> LispObject {
+    if position.is_marker() {
+        set_point_from_marker(position);
+    } else if position.is_fixnum() {
```
Comment (birkenfeld) [inline rust_src/src/editfns.rs]: -> pattern match `as_fixnum()`

File: rust_src/src/frame.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//! Generic frame functions.
```
Comment (birkenfeld) [inline rust_src/src/frame.rs]: file should be `frames.rs` to be consistent.

File: rust_src/src/marker.rs [hunk 3]
```diff
@@ -70,3 +80,23 @@ pub fn marker_buffer(marker: LispObject) -> LispObject {
         None => LispObject::constant_nil(),
     }
 }
+
+/// Set PT from MARKER's clipped position.
+pub fn set_point_from_marker(marker: LispObject) {
```
Comment (birkenfeld) [inline rust_src/src/marker.rs]: could this take a LispMarkerRef instead?

File: rust_src/src/marker.rs [hunk 4]
```diff
@@ -70,3 +80,23 @@ pub fn marker_buffer(marker: LispObject) -> LispObject {
...
+    let m = marker.as_marker_or_error();
+    let charpos = clip_to_bounds(
+        cur_buf.begv,
+        marker_position(marker).as_fixnum().unwrap(),
+        cur_buf.zv,
+    );
+    let mut bytepos = m.bytepos_or_error();
+    // Don't trust the byte position if the marker belongs to a
+    // different buffer.
+    if m.buffer().map_or(false, |b| b.as_ptr() != cur_buf.as_ptr()) {
```
Comment (birkenfeld) [inline rust_src/src/marker.rs]: I wonder if we should add an impl to avoid the `as_ptr` here.

File: rust_src/src/marker.rs [hunk 5]
```diff
@@ -70,3 +80,23 @@ pub fn marker_buffer(marker: LispObject) -> LispObject {
...
     }
 }
+
+/// Set PT from MARKER's clipped position.
+pub fn set_point_from_marker(marker: LispObject) {
+    let cur_buf = ThreadState::current_buffer();
+    let m = marker.as_marker_or_error();
+    let charpos = clip_to_bounds(
+        cur_buf.begv,
+        marker_position(marker).as_fixnum().unwrap(),
```
Comment (birkenfeld) [inline rust_src/src/marker.rs]: `maker_position` just wraps charpos in a lisp_object that you unwrap again here :)

Comment (brotzeit) [reply rust_src/src/marker.rs]: I tried it without, but it doesn't work.

File: rust_src/src/frame.rs (hunk 2 above)
Comment (brotzeit) [reply rust_src/src/frame.rs]: I forgot this one. I'll change this tomorrow

**Comment by birkenfeld** [comment]
Just needs a rebase...

**Pull request closed without merging**